client/flutter/lib/components/themed_text.dart

```diff
@@ -0,0 +1,187 @@
+import 'package:auto_size_text/auto_size_text.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:who_app/constants.dart';
+
+enum TypographyVariant { title, h1, h2, h3, h4, body, button, bodySmall }
+
+class ThemedText extends StatelessWidget {
+  final String data;
+  final TypographyVariant variant;
+
+  final Locale locale;
+  final int maxLines;
+  final TextOverflow overflow;
+  final String semanticsLabel;
+  final bool softWrap;
+  final StrutStyle strutStyle;
+  final TextStyle style;
+  final TextAlign textAlign;
+  final TextDirection textDirection;
+  final double textScaleFactor;
+  final TextWidthBasis textWidthBasis;
+
+  ThemedText(
+    this.data, {
+    Key key,
+    @required this.variant,
+    this.locale,
+    this.maxLines,
+    this.overflow,
+    this.semanticsLabel,
+    this.softWrap,
+    this.strutStyle,
+    this.style,
+    this.textAlign,
+    this.textDirection,
+    this.textScaleFactor,
+    this.textWidthBasis,
+  }) : super(key: key);
+
+  static final Map<TypographyVariant, TextStyle> variantStyles = {
```

Comment:
If we use a function containing a switch statement with all the `TypographyVariant ` values then as long as it doesn't contain the `default` clause, a warning is created. A map doesn't enforce all enum values to be specified.

Reply:
Good point. I'll use a function & switch instead